cmd/livepeer/starter/starter.go

```diff
@@ -1201,6 +1058,174 @@ func StartLivepeer(ctx context.Context, cfg LivepeerConfig) {
 		}()
 	}
 
+	var aiCaps []core.Capability
+	capabilityConstraints := make(map[core.Capability]*core.PerCapabilityConstraints)
+
+	if *cfg.AIWorker {
+		gpus := []string{}
+		if *cfg.Nvidia != "" {
+			var err error
+			gpus, err = common.ParseAccelDevices(*cfg.Nvidia, ffmpeg.Nvidia)
+			if err != nil {
+				glog.Errorf("Error parsing -nvidia for devices: %v", err)
+				return
+			}
+		}
+
+		modelsDir := *cfg.AIModelsDir
+		if modelsDir == "" {
+			var err error
+			modelsDir, err = filepath.Abs(path.Join(*cfg.Datadir, "models"))
+			if err != nil {
+				glog.Error("Error creating absolute path for models dir: %v", modelsDir)
+				return
+			}
+		}
+
+		if err := os.MkdirAll(modelsDir, 0755); err != nil {
+			glog.Error("Error creating models dir %v", modelsDir)
+			return
+		}
+
+		n.AIWorker, err = worker.NewWorker(*cfg.AIRunnerImage, gpus, modelsDir)
+		if err != nil {
+			glog.Errorf("Error starting AI worker: %v", err)
+			return
+		}
+
+		if *cfg.AIModels != "" {
+			configs, err := core.ParseAIModelConfigs(*cfg.AIModels)
+			if err != nil {
+				glog.Errorf("Error parsing -aiModels: %v", err)
+				return
+			}
+
+			for _, config := range configs {
+				modelConstraint := &core.ModelConstraint{Warm: config.Warm}
+
+				// Set price per unit base info.
+				pixelsPerUnit, ok := new(big.Rat).SetString(*cfg.PixelsPerUnit)
+				if !ok || !pixelsPerUnit.IsInt() {
+					panic(fmt.Errorf("-pixelsPerUnit must be a valid integer, provided %v", *cfg.PixelsPerUnit))
+				}
+				if pixelsPerUnit.Sign() <= 0 {
+					// Can't divide by 0
+					panic(fmt.Errorf("-pixelsPerUnit must be > 0, provided %v", *cfg.PixelsPerUnit))
+				}
+				pricePerUnit, currency, err := parsePricePerUnit(config.PricePerUnit.String())
```

Comment:
The price should be specified as a separate field in the `config`, not as a string. See the logic for per-gateway prices: https://github.com/livepeer/go-livepeer/blob/8036856cbfd6724f79b393118e49fd2451282797/cmd/livepeer/starter/starter.go#L820-L826

This is also related to this discussion: https://github.com/livepeer/go-livepeer/pull/3103/commits/6292df161f1c719e42eab42af152b0b748b32a1a#r1698466441